### PR TITLE
Remove lodash from repofs source code

### PR DESCRIPTION
**Pull request by mklabs**
This PR removes the dependency lodash from the actual source code (while it is still used in the tests)

The bundle size is decreased by approximatively 500kb.

```
-rw-r--r--  1 mdnl  staff   1.1M Dec 12 12:56 with-lodash.js
-rw-r--r--  1 mdnl  staff   648K Dec 12 12:54 without-lodash.js
```

with gzip

```
cat with-lodash.js | gzip | wc -c
  224949
cat without-lodash.js | gzip | wc -c
  129901
```

which represents a net 100kb gain with gzip.

### CODE REVIEW

File: src/utils/arraybuffer.js [hunk 1]
```diff
@@ -77,7 +76,7 @@ function enforceArrayBuffer(b, encoding) {
  * Force conversion to string with specific encoding
  */
 function enforceString(b, encoding) {
-    if (_.isString(b)) return b;
+    if (typeof b === 'string') return b;
```
Comment (SamyPesse) [inline src/utils/arraybuffer.js]: We can maybe use [`is`](https://www.npmjs.com/package/is) ?

File: src/utils/mapValues.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/**
```
Comment (SamyPesse) [inline src/utils/mapValues.js]: Why not use [`map-obj`](https://www.npmjs.com/package/map-obj) ?

File: src/utils/fromPairs.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/**
```
Comment (SamyPesse) [inline src/utils/fromPairs.js]: I'd like to avoid this kind of utility here. We can publish it as an node module or use an existing one.

Comment (Soreine) [inline src/utils/fromPairs.js]: Unless we find an existing module, let's just replace the single occurrence of this utility with:
```js
var result = pairs.reduce((result, [key, value]) => {
  result[key] = value;
  return result;
}, {});
```

File: src/utils/mapValues.js (hunk 2 above)
Comment (mklabs) [inline src/utils/mapValues.js]: Used [modify-values](https://www.npmjs.com/package/modify-values) instead.

**Pull request merged**